src/test/java/internship/framework/pages/PersonalDetailsPage.java

```diff
@@ -0,0 +1,191 @@
+        ProjectLogger.info("Select title.");
+        Select select = new Select(selectTitleElement);
+        select.selectByVisibleText("Mr");
+        return this;
+    }
+
+    public PersonalDetailsPage fillWithValueFirstName() {
+        ProjectLogger.info("Fill first name element with value.");
+        ElementUtils.fillInWithValue(firstNameElement, TestData.YOUR_DETAILS_FIRST_NAME);
+        return this;
+    }
+
+    public PersonalDetailsPage fillWithValueLastName() {
+        ProjectLogger.info("Fill last name element with value.");
+        ElementUtils.fillInWithValue(lastNameElement, TestData.YOUR_DETAILS_LAST_NAME);
+        return this;
+    }
+
+    public PersonalDetailsPage fillWithValueAddress() {
+        ProjectLogger.info("Fill address element with value.");
+        ElementUtils.fillInWithValue(addressElement, TestData.YOUR_DETAILS_ADDRESS);
+        return this;
+    }
+
+    public PersonalDetailsPage fillWithValueCityTown() {
+        ProjectLogger.info("Fill city/town element with value.");
+        ElementUtils.fillInWithValue(cityTownElement, TestData.YOUR_DETAILS_CITY_TOWN);
+        return this;
+    }
+
+    public PersonalDetailsPage selectCountry() {
+        ProjectLogger.info("Select country.");
+        Select select = new Select(countryElement);
+        select.selectByVisibleText("Romania");
+        return this;
+    }
+
+    public PersonalDetailsPage fillWithValueTelephone() {
+        ProjectLogger.info("Fill telephone element with value.");
+        ElementUtils.fillInWithValue(telephoneElement, TestData.YOUR_DETAILS_TELEPHONE);
+        return this;
+    }
+
+    public boolean isValidateChangesDisplayed() {
+        ProjectLogger.info("Check if validate changes element is displayed.");
+        return ElementUtils.doesAppear(validateChangesElement);
+    }
+
+    public void clickSaveChangesButton() {
+        ProjectLogger.info("Save changes made.");
+        ElementUtils.clickOn(saveChangesElement);
+    }
+
+    public PersonalDetailsPage selectAllContactPreferences() {
+        ProjectLogger.info("We check all contact preferences.");
+        ElementUtils.clickOn(firstCheckBoxElement);
+        ElementUtils.clickOn(secondCheckBoxElement);
+        ElementUtils.clickOn(thirdCheckBoxElement);
+        ElementUtils.clickOn(fourthCheckBoxElement);
+        return this;
+    }
+
+    public PersonalDetailsPage navigateSubscriptionDetailsTab() {
+        ProjectLogger.info("Navigate to Subscription Details page.");
+        ElementUtils.clickOn(subscriptionDetailsButton);
+        return this;
+    }
+
+    public void navigatePersonalDetailsTab() {
+        ProjectLogger.info("Navigate to Personal Details page.");
+        ElementUtils.clickOn(personalDetailsButton);
+    }
+
+    private boolean isFirstCheckBoxSelected() {
+        ProjectLogger.info("Check if first contact checkbox is selected.");
+        Driver.get().manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
```

Comment:
Why do we do this only for the first checkbox ?

Reply:
I need the implicit wait time between navigating to Personal details page and checking the first checkbox. I moved that line inside test method.